### ISSUE TITLE
mixout 输出

### ISSUE THREAD

**Issue by jinyuhang** [bloodwass/mixout#6]
在MIxOut类中，这个计算方式没看懂
if ctx.p == 1:
            output = target.clone()
        else:
            output = ((1 - ctx.noise) * target + ctx.noise * output - ctx.p * target) / (1 - ctx.p)
        return output

论文里貌似也不是这样的描述，不太懂，能不能给解释一下

**Comment by bloodwass** [comment]
I cannot read your question.

But according to Google translator,

this link would help you
 
[link](https://github.com/bloodwass/mixout/issues/5#issuecomment-772694638)

**Comment by jinyuhang** [comment]
> I cannot read your question.
> 
> But according to Google translator,
> 
> this link would help you
> 
> [link](https://github.com/bloodwass/mixout/issues/5#issuecomment-772694638)

thanks very much!

**Comment by jinyuhang** [comment]
I have read the content,

_It is a rescale technique.
This line makes the expectation of the mixed parameters,
E[(1 - Bernoulli(1-p)) * target + Bernoulli(1-p) * input] = p*target + (1-p)*input,
be full parameters, input._

so, the calculation makes the expectation, I can understand it.
 But i wonder the reason that parameters and  input is added?
**Can the parameters be a part of output?**

**Comment by bloodwass** [comment]
In this case, 'target', which is the set of initial parameters, is not trainable (it is not updated by gradient descent). In the case of dropout, 'target' is the set of all zero parameters.

**Comment by jinyuhang** [comment]
'target', which is the set of initial parameters from pretrain model. 

Although it is not trainable, But what's the point of making it part of the output?
what's the goal??

**Comment by bloodwass** [comment]
For dropout, you can implement it by masking inputs with Bernoulli random variables.

For example, if you want to apply dropout to a fully connected layer with a weight matrix W and an input x, you can directly mask x and compute W * mask(x).

However, I have no idea to implement mixout like dropout.

Instead, I use the fact that "masking the i-th component of x and matrix multiplying it by W, W * mask_i(x)" is equal to "masking the i-th column of W, mask_i(W), and compute mask_i(W) * x.

That's why my implementation dealt with parameters.

**Comment by jinyuhang** [comment]
您好，邮件你的我已经收到！我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
